From d80a838e81a19d5d9c9cee204b6bf8e432775516 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Wed, 23 Jul 2025 13:22:10 +0200
Subject: [PATCH] sd: Fix no matching function for call on armhf
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

sd/qa/unit/tiledrendering/tiledrendering.cxx: In member function ‘void testSlideshowLayeredRendering::TestBody()’:
sd/qa/unit/tiledrendering/tiledrendering.cxx:3033:5: error: no matching function for call to ‘assertEquals(int, sal_Int32&, CppUnit::SourceLine, const char [1])’
 3033 |     CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
      |     ^
/usr/include/cppunit/TestAssert.h:161:6: note: candidate: ‘template<class T> void CppUnit::assertEquals(const T&, const T&, SourceLine, const std::string&)’
  161 | void assertEquals( const T& expected,
      |      ^~~~~~~~~~~~
/usr/include/cppunit/TestAssert.h:161:6: note:   template argument deduction/substitution failed:
sd/qa/unit/tiledrendering/tiledrendering.cxx:3033:5: note:   deduced conflicting types for parameter ‘const T’ (‘int’ and ‘sal_Int32’ {aka ‘long int’})
 3033 |     CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
      |     ^

Change-Id: Ie6c0f59207a9e953b491f4a1ac8a8a201efcea76
---
 sd/qa/unit/tiledrendering/tiledrendering.cxx | 28 ++++++++++----------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/sd/qa/unit/tiledrendering/tiledrendering.cxx b/sd/qa/unit/tiledrendering/tiledrendering.cxx
index 7059c59391ca..dedb622cd55a 100644
--- a/sd/qa/unit/tiledrendering/tiledrendering.cxx
+++ b/sd/qa/unit/tiledrendering/tiledrendering.cxx
@@ -3030,8 +3030,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering)
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     const Color aTransparentColor(ColorAlpha, 0x00000000);
     const tools::Long nLeft = 20;
@@ -3148,8 +3148,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_WithFie
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     const Color aTransparentColor(ColorAlpha, 0x00000000);
     const tools::Long nLeft = 20;
@@ -3528,8 +3528,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -3700,8 +3700,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
     sal_Int32 nViewHeight = 2000;
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -3962,8 +3962,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -4129,8 +4129,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1125), nViewHeight);
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -4234,8 +4234,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1500, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(2000), nViewWidth);
+    CPPUNIT_ASSERT_EQUAL(sal_Int32(1500), nViewHeight);
 
     std::vector<std::string> aBufferHashes;
 
-- 
2.50.0

